### ISSUE TITLE
Gas unit_of_measurement not reported correctly

### ISSUE THREAD

**Issue by Twincarb** [scooter1556/glow-mqtt#2]
I am getting my data from Glowmarkt MQTT broker and the unit of measurement is displayed as m³ whereas the data provided is kWh, can an option be added to allow the correct selection rather than:
`   gas_unit = "kWh" if local else "m³"`
I am not sure if this is due to my meter being SMETS 1?

**Comment by Twincarb** [comment]
I have just had a look at the data which is returned in the raw and compared to what is written on the following line https://gist.github.com/ndfred/b373eeafc4f5b0870c1b8857041289a9#file-glowmqtt-py-L21 
Is it possible to determine the correct unit of measure based on the first mqtt data message recieved?

**Comment by scooter1556** [comment]
Hi @Twincarb,

I'm a bit blind on this one as I don't have a gas meter, I've only been working on reference data provided by others.

Does the following match what you see?

`data['gasMtr']['0702']['00']['00'] == KWh`
`data['gasMtr']['0702']['00']['01'] == m3`

Today the Home Assistant MQTT discovery pre-populates before any messages are received. I would need to refactor this to auto-detect and create the correct sensors which isn't a bad idea anyway for those who only have electric for example...

**Comment by Twincarb** [comment]
Hi Scooter,  below is the full gas data that I receive, which I think means:

`data['gasMtr']['0702']['03']['00'] == KWh`
`data['gasMtr']['0702']['03']['01'] == m3`

```
"gasMtr":
{
	"0702":
	{
		"03": 
		{
			"01":"000001",
			"05":"43",
			"04":"9B",
			"02":"0003E8",
			"07":"***********",
			"08":"***********",
			"03":"43",
			"00":"00",
			"06":"80"
		},
		"00":
		{
			"07":"2A8C6569",
			"01":"000000000000",
			"00":"0000062A4616",
			"14":"00",
			"02":"000000000000"
		},
		"04":
		{
			"01":"001BF8",
			"00":"000000"
		},
		"02":
		{
			"00":"00"
		}
	}
},
```
Any way I can help let me know

**Comment by scooter1556** [comment]
Thanks @Twincarb, that helps.

Let me see what I can do :)

**Comment by scooter1556** [comment]
I have updated the child project with the functionality mentioned. Meters present and gas units should be auto-detected on first message and used to configure Home Assistant. I will update the addon later today but feel free to try the Python app manually and let me know if there are any issues:

https://github.com/scooter1556/glow-mqtt/commit/7ff55358eeaeb9cd222f3736b07b23095cc04f8c

**Comment by Twincarb** [comment]
Thanks Scot, I updated last night and the data kept coming which confirms it hasn't broken that. I still show kWh I suspect because I already have data stored under the original m3. I tried using SERVICE PURGE_ENTITIES but that appears not to remove the data. 
I am away from home now for the next couple of days but when I am back I will try on a clean install. Certainly reading through the code I can't see any issues.
If you have any suggestions on the removal of the data from the old entity that would be helpful although not essential as I had planned a clean reinstall in the near future!
